gemini/master.py

```diff
@@ -13,11 +15,48 @@
 NODE_TOKEN_BYTES = 32
 NODE_PASSWORD_BYTES = 32
 
+STORAGE_TYPE = "memory"
+
 app = Flask('gemini-master')
 
 c = consul.Consul(host='127.0.0.1', port=8500)
 
-jobs = []
+jobs: List[Any] = []
+
+
+class MemoryNodeInfoRepository:
+    def __init__(self):
+        self.node_data = {}
+
+    def put_data(self, value: dict):
+        self.node_data[value['node_id']] = value
+
+    def get_data(self, node_id: str) -> dict:
+        return self.node_data[node_id]
+
+
+class ConsulNodeInfoRepository:
+    def __init__(self):
+        self.node_data = {}
+
+    def put_data(self, value: dict):
+        c.kv.put('nodes/' + value['node_id'], json.dumps(value))
+
+    def get_data(self, node_id: str) -> dict:
```

Comment:
Maybe this should return a more specific type. Like `Mapping[str, Any]` or if we know more about the returned structure we can return a customly defined `TypedDict`